plugins/generic/externalFeed/classes/ExternalFeedDAO.inc.php

```diff
@@ -1,45 +1,59 @@
 <?php
 
 /**
- * @file plugins/generic/externalFeed/ExternalFeedDAO.inc.php
+ * @file plugins/generic/externalFeed/classes/ExternalFeedDAO.inc.php
  *
  * Copyright (c) 2014-2018 Simon Fraser University
  * Copyright (c) 2003-2018 John Willinsky
  * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
  *
  * @class ExternalFeedDAO
- * @ingroup plugins_generic_externalFeed
+ * @ingroup plugins_generic_externalFeed_classes
  *
  * @brief Operations for retrieving and modifying ExternalFeed objects.
  */
 
 import('lib.pkp.classes.db.DAO');
 
 class ExternalFeedDAO extends DAO {
-	/** @var string Name of parent plugin */
-	var $parentPluginName;
+	/** @var ExternalFeedPlugin reference to ExternalFeed plugin */
+	protected $parentPlugin = null;
 
 	/**
 	 * Constructor
 	 */
-	function __construct($parentPluginName) {
-		$this->parentPluginName = $parentPluginName;
+	public function __construct($parentPlugin) {
+		$this->parentPlugin = $parentPlugin;
 		parent::__construct();
 	}
 
+	/**
+	 * Instantiate a new data object.
+	 * @return ExternalFeed
+	 */
+	function newDataObject() {
+		$this->parentPlugin->import('classes.ExternalFeed');
+		return new ExternalFeed();
+	}
+
 	/**
 	 * Retrieve an ExternalFeed by ID.
 	 * @param $feedId int
+	 * @param $contextId int 
 	 * @return ExternalFeed
 	 */
-	function &getExternalFeed($feedId) {
+	public function getById($feedId, $contextId = null) {
+		$params = array((int) $feedId);
+		if ($contextId) $params[] = (int) $contextId;
+
 		$result = $this->retrieve(
-			'SELECT * FROM external_feeds WHERE feed_id = ?', $feedId
+			'SELECT * FROM external_feeds WHERE feed_id = ? ' . ($contextId?' AND journal_id = ?':''),
```

Comment:
Again, this still has some OJS-specific language when I think it could easily be made to work verbatim with OMP as well. Search and replace throughout (https://github.com/pkp/ojs/pull/1115#discussion_r162739683).

Reply:
So the table column should be `context_id` instead of `journal_id`.  right?

Reply:
Yes, that's right. We don't have a good toolset for upgrade scripts within plugins, so it's something to consider before the plugin is released -- maybe file this in the new repo for some more specific consideration?

Reply:
https://github.com/kaschioudi/externalFeed/issues/1